lizmap/modules/lizmap/lib/Request/RemoteStorageRequest.php

```diff
@@ -0,0 +1,361 @@
+ */
+
+namespace Lizmap\Request;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7;
+
+class RemoteStorageRequest
+{
+    protected static $webDAVNamespace = 'DAV:';
+
+    protected static $fileNameExpression = 'file_name(@selected_file_path)';
+    protected static $appContext;
+
+    /**
+     * @param string $storageUrl remote destination path
+     * @param string $file       the file path to upload
+     *
+     * @return array{0: null|string, 1: int, 2: string} Array url of the uploaded file (0: string), HTTP code (1: int), message (2: string)
+     */
+    public static function uploadToWebDAVStorage($storageUrl, $file)
+    {
+        $resource = null;
+        $http_code = 400;
+        $message = null;
+        $returnUrl = null;
+        // checks on file
+        if (!is_file($file)) {
+            \jLog::log($file.' The file is not valid', 'error');
+
+            return array(null, 400, 'The file is not valid');
+        }
+
+        try {
+            $resource = Psr7\Utils::tryFopen($file, 'r');
+        } catch (\RuntimeException $e) {
+            \jLog::log($e->getMessage(), 'error');
+
+            return array(null, 400, 'Error on file upload');
+        }
+
+        // get profile connection parameters
+        $profile = self::getProfile('webdav');
+
+        if ($profile) {
+            // baseUri check
+            if (strpos($storageUrl, $profile['baseUri']) === 0) {
+                $opt = array();
+                $headers = array();
+                $headers['User-Agent'] = 'lizmap-user-agent';
+
+                if (array_key_exists('user', $profile) && array_key_exists('password', $profile)) {
+                    $opt['auth'] = array();
+                    array_push($opt['auth'], $profile['user']);
+                    array_push($opt['auth'], $profile['password']);
+                }
+                $opt['headers'] = $headers;
+
+                $stream = \GuzzleHttp\Psr7\Utils::streamFor($resource);
+                $opt['body'] = $stream;
+
+                $client = new Client();
+
+                try {
+                    $response = $client->request('PUT', $storageUrl, $opt);
+                    $returnUrl = $storageUrl;
+                    $http_code = $response->getStatusCode();
+                } catch (\GuzzleHttp\Exception\RequestException $e) {
+                    $message = 'Error on file upload '.$e->getMessage();
+                    if ($e->hasResponse()) {
+                        $http_code = $e->getResponse()->getStatusCode();
+                    }
+                } catch (\Exception $e) {
+                    $http_code = 500;
+                    $message = 'Error on file upload '.$e->getMessage();
+                }
+            } else {
+                $message = 'Invalid path '.$storageUrl;
+            }
+        } else {
+            $message = 'WebDAV configuration not found';
+        }
+        if ($message) {
+            \jLog::log($message, 'error');
```

Comment:
you can use `$this->appContext->logMessage`

Reply:
OK thank you. Actually in most cases the method returns the error message so I will remove the logs inside the method and let the caller log the error or not